vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -486,9 +487,23 @@ import { ItemCache } from '@vaadin/vaadin-grid/src/vaadin-grid-data-provider-mix
                 .forEach(idx => delete parentCache.itemCaches[idx]);
           }
           delete lastRequestedRanges[parentKey];
-          this.collapseItem(inst.item);
+          this.collapseItem(item);
         }
-      })
+      });
+
+      // Monkey-patch grid.expandItem and grid.collapseItem to have
+      // itemExpandedChanged run when either happens.
+      const originalExpandItem = grid.expandItem;
+      grid.expandItem = (item) => {
```

Comment:
I looked over the connector again and found that `_updateItem` is also overridden so that I would suggest aligning the way they are overridden.

```js
grid._updateItem = tryCatchWrapper(function(row, item) { // <--- doesn't use arrow function
  GridElement.prototype._updateItem.call(grid, row, item); // <--- calls `_updateItem` from prototype
});
```

Reply:
Good idea. Done.